src/Foundation/Base/code/ModelMappers/Mapper.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace Wooli.Foundation.Base.ModelMappers
+{
+    using System.Reflection;
+
+    using AutoMapper;
+    using AutoMapper.Configuration;
+
+    using Sitecore.Diagnostics;
+
+    public class Mapper : IMapper
+    {
+        protected readonly AutoMapper.IMapper InnerMapper;
+
+        public Mapper(Assembly assembly)
+        {
+            Assert.ArgumentNotNull(assembly, nameof(assembly));
+
+            var mappings = this.GetMappings(assembly);
+            var config = new MapperConfiguration(mappings);
+
+            this.InnerMapper = new AutoMapper.Mapper(config);
+        }
+
+        public Mapper()
+            : this(Assembly.GetCallingAssembly())
+        { }
+
+        public TResult Map<TSource, TResult>(TSource source)
+        {
+            return this.InnerMapper.Map<TSource, TResult>(source);
+        }
+
+        private MapperConfigurationExpression GetMappings(Assembly assembly)
+        {
+            var baseMappings = new MapperConfigurationExpression();
+            baseMappings.AddProfiles(assembly);
```

Comment:
We discussed with you issue with auto-registration and DI here. We can also separate it as we want, like it already done in Foundation.Commerce
```
public abstract class Mapper
    {
        protected abstract readonly IMapper mapper;
        protected abstract MapperConfiguration configuration {get;set;}

        public Mapper()
        { 
        }

        public TDestination Map<TSource, TDestination>(TSource source)
        {
            return this.mapper.Map<TSource, TDestination>(source);
        }
    }
```
```
public CategoryMapper : Mapper, ICategoryMapper
{
            ///...
         MapperConfiguration configuration = new ...
         this.mapper = new Mapper(configuration);
}
```
something like that, no profiles